src/index.ts

```diff
@@ -56,7 +58,8 @@ export = {
         '@typescript-eslint/explicit-function-return-type': ['error', {
           allowExpressions: true,
           allowHigherOrderFunctions: true,
-          allowTypedFunctionExpressions: true
+          allowTypedFunctionExpressions: true,
+          allowDirectConstAssertionInArrowFunctions: true
```

Comment:
https://github.com/typescript-eslint/typescript-eslint/pull/876
I feel that it's a welcome relaxation of the rule. I don't see any downside to it.